go/pools/resource_pool.go

```diff
@@ -311,7 +435,14 @@ func (rp *ResourcePool) SetCapacity(capacity int) error {
 
 	if capacity < oldcap {
 		for i := 0; i < oldcap-capacity; i++ {
-			wrapper := <-rp.resources
+			var wrapper resourceWrapper
```

Comment:
I would love a comment that explains the following code. It looks clever and interesting, but not obvious